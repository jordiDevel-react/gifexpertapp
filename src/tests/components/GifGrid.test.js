import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid /> ', () => {
  const category = 'Dragon Ball';

  test('debe mostrar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar items cuando se cargan las imÃ¡genes', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa',
      },
      {
        id: 'def',
        url: 'https://localhost/cualquier/otracosa.jpg',
        title: 'Otra cosa',
      },
    ];
    useFetchGifs.mockReturnValue({ data: gifs, loading: false });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
