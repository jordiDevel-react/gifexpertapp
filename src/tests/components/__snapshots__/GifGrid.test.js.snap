// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid />  debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid />  debe mostrar items cuando se cargan las im√°genes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="def"
      key="def"
      title="Otra cosa"
      url="https://localhost/cualquier/otracosa.jpg"
    />
  </div>
</Fragment>
`;
